src/Tribe/Views/V2/Template_Bootstrap.php

```diff
@@ -228,6 +228,13 @@ public function get_view_html() {
 		$should_display_single = apply_filters( 'tribe_events_views_v2_bootstrap_should_display_single', $should_display_single, $view_slug, $query, $context );
 
 		if ( $should_display_single ) {
+			// Set our context to read as a single-event view.
+			add_filter(
+				"tribe_context_pre_view",
+				function () {
+					return 'single-event';
+				}
+			);
```

Comment:
This doesn't feel like the correct place to do this at all.

Reply:
@bordoni I agree, but what would you recommend? The flag of whether this is a "single event" is determined here, and without overriding the context the `view` slug still says `list`.

Reply:
A couple of things:
* Avoid anonymous functions of this nature on production code they can't be removed.
* Move the adding *and removing* of the filter to inside of the `get_v1_single_event_html` which is where we handle all the legacy V1 single problems so the code makes a lot more sense there.
* Potentially use `tribe_context_view` instead of the `pre` filter.